src/services/socketIo.service.js

```diff
@@ -0,0 +1,41 @@
+const io = require('socket.io')();
+const redis = require('socket.io-redis');
+const config = require('config');
+const { getSocketRoomNameForUser } = require('resources/user/user.helpers');
+const { decodeToken } = require('auth.service');
+
+io.adapter(redis({
+  host: config.redis.host,
+  port: config.redis.port,
+}));
+
+io.use((socket, next) => {
+  const { token } = socket.handshake.query;
+  const decodedToken = decodeToken(token);
+
+  if (decodedToken) {
+    // eslint-disable-next-line no-param-reassign
+    socket.handshake.data = {
+      userId: decodedToken._id,
+    };
+
+    return next();
+  }
+
+  return next(new Error('token is invalid'));
+});
+
+io.on('connection', (client) => {
+  client.on('subscribe', () => {
+    const { userId } = client.handshake.data;
+    const room = getSocketRoomNameForUser(userId);
```

Comment:
@Mar1nka please remove `getSocketRoomNameForUser` helper from server and subscribe to user room from the client side instead (client side: `socket.emit('subscribe', 'user-${user._id}')`

On the server side, prior to joining `user-${user._id}` room we need to validate that current user has an access to it (in this case check that `handshake.data.userId === userId`